src/hal/utils/halcmd_commands.c

```diff
@@ -1640,16 +1640,16 @@ static void print_comp_info(char **patterns)
     while (next != 0) {
 	comp = SHMPTR(next);
 	if ( match(patterns, comp->name) ) {
-            if(comp->type == 2) {
+            if(comp->type == COMPONENT_TYPE_OTHER) {
                 hal_comp_t *comp1 = halpr_find_comp_by_id(comp->comp_id & 0xffff);
                 halcmd_output("    INST %s %s",
                         comp1 ? comp1->name : "(unknown)", 
                         comp->name);
             } else {
                 halcmd_output(" %5d  %-4s  %-*s",
-                    comp->comp_id, (comp->type ? "RT" : "User"),
+                    comp->comp_id, (comp->type == COMPONENT_TYPE_REALTIME) ? "RT" : "User",
```

Comment:
there is a functional change here, old code checks > 0 new code == 1

Reply:
but it does the correct thing, so I do approve this PR, lets wait for the tests to run.

Reply:
Well spotted. I did wonder which would be best, and decided that this was probably what was desired.